### PR TITLE
Upgrade paver to brink; pin dep versions; add build to .gitignore - closes #80

### LINKED ISSUE #80: Fix `paver` script; migrate it to brink.sh
> @adiroiban : I noticed that `paver generate` no longer works, precisely because versions are not pinned (like in my previous CI pull request).

The paver.sh needs to be migrated to brink.sh (see chevah/server brink.sh and brink.conf)
Try to work with @dumol to understand how we bootstrap the python virtualenv using brink.sh

@dumol I am not sure where we have to documentation for brink.sh , so this can be a good opportunity to create the reference documentation.,

_Originally posted by @adiroiban in https://github.com/chevah/geck/issues/79#issuecomment-769772349_

### PR DESCRIPTION
**Pull request by danuker**
Scope
=====

Use `brink` instead of `paver`. Building the docs now works.

Why we got into this (5 whys)
=============================

`paver.sh generate` stopped working because versions were not pinned.

Changes
=======

Deleted `paver.sh`, used `brink` from `server` instead.

Pinned the versions using the `pypi.chevah.com` index.

Added `build` (the HTML output) to `.gitignore`.

How to try and test the changes
===============================

reviewers: @adiroiban 

Run `brink.sh generate`, click the resulting link.

### CODE REVIEW

**Comment by danuker** [comment]
needs-review

File: brink.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BASE_REQUIREMENTS='pip==20.2.4 chevah-brink==0.78.1 paver==1.2.4'
+PYTHON_CONFIGURATION='default@2.7.18.bd461724:sol10@2.7.8.bd461724'
+BINARY_DIST_URI='http://binary.chevah.com/production'
+PIP_INDEX='http://pypi.chevah.com/simple'
+CHEVAH_BUILD_DIR='build-server'
```
Comment (adiroiban) [inline brink.conf:5]: I think that that just "build" ... or delete this line and leave the default config.

```suggestion
CHEVAH_BUILD_DIR='build'
```

File: brink.conf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+BASE_REQUIREMENTS='pip==20.2.4 chevah-brink==0.78.1 paver==1.2.4'
```
Comment (adiroiban) [inline brink.conf:1]: We don't need chevah-brink here... only pip and paver

```suggestion
BASE_REQUIREMENTS='pip==20.2.4 paver==1.2.4'
```

File: pavement.py [hunk 3]
```diff
@@ -10,9 +10,9 @@
 
 
 RUN_PACKAGES = [
-    'sphinx',
-    'sphinx-autobuild',
-    'sphinx_rtd_theme',
+    'sphinx==1.8.5',
```
Comment (adiroiban) [inline pavement.py:13]: Maybe update to latest Sphinx ?

File: brink.conf [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+PYTHON_CONFIGURATION='default@2.7.18.bd461724:sol10@2.7.8.bd461724'
```
Comment (adiroiban) [inline brink.conf:2]: This is ok with simple default...we don't plan to use it on solaris 10 :)

```suggestion
PYTHON_CONFIGURATION='default@2.7.18.bd461724'
```

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks.

I will try to see with Read The Docs (RTD) build was not trigerred.

File: pavement.py (hunk 3 above)
Comment (danuker) [reply pavement.py:13]: Sphinx dropped Python 2 support in 2018; should I also update to Python 3?

https://github.com/sphinx-doc/sphinx/commit/9412bd76b7f5fdf0adc231ef8130e45bda130164

File: brink.conf (hunk 1 above)
Comment (danuker) [reply brink.conf]: Hmm, I think a better name for that directory is perhaps "env"; what do you think?

File: pavement.py (hunk 3 above)
Comment (adiroiban) [reply pavement.py:13]: Yes please :)

For Python3 we should use https://github.com/chevah/pythia/

If need help on using pythia let me know or @dumol :)

 pythia is our future py3 own distribution and Mișu is maintaining it

**Comment by danuker** [comment]
Works on Python 3 with Pythia; but running `./brink.sh generate` a second time yields an error:

```
$ ./brink.sh generate
Couldn't detect a supported distribution. Trying to treat it as generic Linux...
cat: build/lib/PYTHIA_VERSION: No such file or directory
```

Instead of a `PYTHIA_VERSION` file I have `PYTHON_PACKAGE_VERSION` with `3.8.6.e939358` inside.

Did Pythia not get run?

needs-review

**Review by adiroiban — APPROVED** [review]
Nicely done. Thanks!

**Comment by adiroiban** [comment]
> Instead of a `PYTHIA_VERSION` file I have `PYTHON_PACKAGE_VERSION` with `3.8.6.e939358` inside.
> 
> Did Pythia not get run?

Please check with @dumol .... maybe it's a bug. Thanks!

**Comment by dumol** [comment]
> > Instead of a `PYTHIA_VERSION` file I have `PYTHON_PACKAGE_VERSION` with `3.8.6.e939358` inside.
> > Did Pythia not get run?
> 
> Please check with @dumol .... maybe it's a bug. Thanks!

Does a `clean` followed by a `deps` solve this?

I know it's confusing that the file is still named `PYTHON_PACKAGE_VERSION`. I've changed that after last release. Next one would use `PYTHIA_VERSION`. More details at https://github.com/chevah/pythia/pull/20/commits/640147db0eefc92dc505260920548f46a250bde7 and the associated PR: https://github.com/chevah/pythia/pull/20.

**Comment by danuker** [comment]
@dumol Running `clean` followed by `deps` creates a `PYTHON_PACKAGE_VERSION`, but not a `PYTHIA_VERSION`.

Then, when I try to run `generate`:

```
$ ./brink.sh generate
Couldn't detect a supported distribution. Trying to treat it as generic Linux...
cat: build/lib/PYTHIA_VERSION: No such file or directory
```

**Comment by dumol** [comment]
> @dumol Running `clean` followed by `deps` creates a `PYTHON_PACKAGE_VERSION`, but not a `PYTHIA_VERSION`.
> 
> Then, when I try to run `generate`:
> 
> ```
> $ ./brink.sh generate
> Couldn't detect a supported distribution. Trying to treat it as generic Linux...
> cat: build/lib/PYTHIA_VERSION: No such file or directory
> ```

That's because the current release of `pythia` generates `build/lib/PYTHON_PACKAGE`.

But, even if you rename the file as needed, you'll run into more problems, as other stuff requires Python 2.

```
$ mv -v build-pythia/lib/PYTHON_PACKAGE_VERSION build-pythia/lib/PYTHIA_VERSION                                                                  
renamed 'build-pythia/lib/PYTHON_PACKAGE_VERSION' -> 'build-pythia/lib/PYTHIA_VERSION'
$ ./brink.sh generate                                                                 
Couldn't detect a supported distribution. Trying to treat it as generic Linux...
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/dumol/src/pythia/build-pythia/lib/python3.8/site-packages/paver/tasks.py", line 878, in main
    _launch_pavement(args)
  File "/home/dumol/src/pythia/build-pythia/lib/python3.8/site-packages/paver/tasks.py", line 846, in _launch_pavement
    exec(compile(source, environment.pavement_file, 'exec'), mod.__dict__)
  File "pavement.py", line 113
    except IOError, e:
                  ^
SyntaxError: invalid syntax
```

**Comment by danuker** [comment]
@dumol @adiroiban So, is it OK to use it as is right now, and upgrade when Pythia is fully Py3-ready?

**Comment by dumol** [comment]
> @dumol @adiroiban So, is it OK to use it as is right now, and upgrade when Pythia is fully Py3-ready?

For the record, `pythia` **is** the Python 3 to base other stuff on. It's the things that comes on top of `pythia` that are not Python 3 ready. Nobody has worked on that until now… The changes I've done to `brink.sh` to make it work for `pythia` are just a preliminary step.

**Comment by adiroiban** [comment]
it looks like the `pavement.py` from python is not updated to py3.

I have created a separate ticket and assigned it to Dan - https://github.com/chevah/pythia/issues/22

So Pythia (our py3 distribution) is Py3 ready ...but the tools we use on top of it are not py3 ready... but we are working on this  and the fixes should be easy.

**Pull request merged**